### ISSUE TITLE
Value Checker issues warnings on implicit up-cast

### ISSUE THREAD

**Issue by kelloggm** [typetools/checker-framework#1299]
Consider the following code:

```java
void testShort_plus(@IntRange(from = 0) short x) {
    @IntRange(from = 1, to = Short.MAX_VALUE + 1) int y = x + 1;
}
```
It generates the following error:
```
ValueCast.java:9: error: [assignment.type.incompatible] incompatible types in assignment.
        @IntRange(from = 1, to = Short.MAX_VALUE + 1) int y = x + 1;
                                                                ^
  found   : @IntRange(from=-2147483648, to=2147483647) int
  required: @IntRange(from=1L, to=32768L) int
```

The Value Checker will issue an error for this code, because `@IntRange(from = 0)` is implicitly treated as `@IntRange(from = 0, to = Long.MAX_VALUE)`, even when the type being annotated is not a `long`. The Value Checker should fill in the appropriate default max value (or min value, when the specified value is `to`) for ranges when only one value is written by the programmer.

(Test case framework/tests/value/ValueCast.java)
